SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -46,15 +46,41 @@ void USpatialLoadBalanceEnforcer::OnAuthorityIntentComponentUpdated(const Worker
 	}
 }
 
+// This is called whenever this worker becomes authoritative for the ACL component on an entity.
+// It is now this worker's responsibility to make sure that the ACL reflects the current intent,
+// which may have been received before this call.
 void USpatialLoadBalanceEnforcer::AuthorityChanged(const Worker_AuthorityChangeOp& AuthOp)
 {
 	if (AuthOp.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID &&
 		AuthOp.authority == WORKER_AUTHORITY_AUTHORITATIVE)
 	{
+		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(AuthOp.entity_id);
+		if (AuthorityIntentComponent == nullptr)
+		{
+			// TODO(zoning): There are still some entities being created without an authority intent component.
+			// For example, the Unreal created worker entities don't have one. Even though those won't be able to
+			// transition, we should have the intent component on them for completeness.
+		 	UE_LOG(LogSpatialLoadBalanceEnforcer, Log, TEXT("Requested queueing of entity without AuthIntent component. EntityId: %lld"), AuthOp.entity_id);
+			return;
+		}
+
+		const FString* OwningWorkerId = VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(AuthorityIntentComponent->VirtualWorkerId);
+		if (OwningWorkerId != nullptr && *OwningWorkerId == WorkerId)
```

Comment:
I don't think this is correct. The ACL can (and often will) be authoritative on a different worker from the rest of the entity.

Reply:
Yep agreed - gaining auth over ACL shouldn't impact which worker currently has auth over all other components

Reply:
Actually, I think I agree with Sami here, although the fact that this is so confusing is itsself a bit worrying. I'll make the change and see how it tests.

Reply:
I could be misreading things but won't it be a no-op to also queue an ACL write request in the case where we are already set to the owning worker id?

Reply:
A no-op yes, but I'm trying to document all the cases where something happens we're not expecting, so that I can upgrade logs to Warnings without introducing spurious warnings that will break NWX's logs.